### ISSUE TITLE
CA1843 - false positive when explicit invoke with list of tasks ?

### ISSUE THREAD

**Issue by fforjan** [dotnet/roslyn-analyzers#6175]
### Analyzer

**Diagnostic ID**: [ca1843](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1843): `Do not use 'WaitAll' with a single task`

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 6.0.0

### Describe the bug

![image](https://user-images.githubusercontent.com/601810/191539430-2f1781b2-a529-43fe-b0ca-cb3e5e75e5f2.png)
We have a extension method which does a bit more than waiting on a set of task and emit a warning when calling Task.WaitAll

### Steps To Reproduce
Simply create such extension method :
```
  public static void FromResult(this Task<Saved[]> list)
        {
            Task.WaitAll(list);         
        }
```
### Expected behavior
Should not have a CA1843 warning

### Actual behavior
emit CA1843 on the call to Task.WaitAll
